src/git/repo.rs

```diff
@@ -1446,6 +1672,98 @@ impl<'repo> Branch<'repo> {
     }
 }
 
+/// A Git file status indicator.
+/// See https://git-scm.com/docs/git-status#_short_format
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum FileStatus {
+    Unmodified,
+    Modified,
+    Added,
+    Deleted,
+    Renamed,
+    Copied,
+    Unmerged,
+    Untracked,
+    Ignored,
+}
+
+/// The status of a file in the repo.
+#[derive(Debug, PartialEq, Eq)]
+pub struct StatusEntry {
+    /// The status of the file in the index.
+    pub index_status: FileStatus,
+    /// The status of the file in the working tree.
+    pub work_tree_status: FileStatus,
```

Comment:
Let's call this `working_copy_status`, as I think that's the terminology we use elsewhere in the project.